src/Integration/NewConnectedMode/IConfigurationPersister.cs

```diff
@@ -54,44 +55,26 @@ public BindingConfiguration Persist(BoundSonarQubeProject project, SonarLintMode
                 throw new ArgumentNullException(nameof(project));
             }
 
-            var targetDirectory = GetTargetDirectory(bindingMode);
+            var configFilePath = GetConfigFilePath(bindingMode);
 
-            var success = targetDirectory != null &&
-                          solutionBindingDataWriter.Write(targetDirectory, project);
+            var success = configFilePath != null &&
+                          solutionBindingDataWriter.Write(configFilePath, project);
 
-            return success ? CreateBindingConfiguration(targetDirectory, project, bindingMode) : null;
+            // The binding directory is the folder containing the binding config file
+            var bindingConfigDirectory = Path.GetDirectoryName(configFilePath);
+            return success ?
+                BindingConfiguration.CreateBoundConfiguration(project, bindingMode, bindingConfigDirectory) : null;
```

Comment:
Logically the code is the same as before, I just in-lined the old `CreateBindingConfiguration` method since it wasn't adding much value.